platform/api/src/api/error.rs

```diff
@@ -6,25 +6,56 @@ use std::{
 use hyper::{Body, StatusCode};
 use serde_json::json;
 
-use super::macros::make_response;
+use super::{macros::make_response, middleware::auth::AuthError};
+
+pub type Result<T, E = ApiErrorInterface> = std::result::Result<T, E>;
+
+#[derive(thiserror::Error, Debug)]
+pub enum ApiError {
+    #[error("failed to upgrade ws connection: {0}")]
+    WsUpgrade(hyper_tungstenite::tungstenite::error::ProtocolError),
+    #[error("no gql request found")]
+    NoGqlRequest,
+    #[error("failed to parse http body: {0}")]
+    ParseHttpBody(hyper::Error),
+    #[error("invalid ws protocol")]
+    InvalidWsProtocol,
+    #[error("failed to parse gql request: {0}")]
+    ParseGql(#[from] async_graphql::ParseRequestError),
+    #[error("http method not allowed")]
+    HttpMethodNotAllowed,
+    #[error("failed to auth: {0}")]
+    Auth(#[from] AuthError),
+    #[error("internal server error: {0}")]
+    InternalServerError(&'static str),
+}
 
-pub type Result<T, E = RouteError> = std::result::Result<T, E>;
+impl From<ApiError> for hyper::Response<Body> {
+    fn from(value: ApiError) -> Self {
+        let status = match &value {
+            ApiError::WsUpgrade(_) => StatusCode::BAD_REQUEST,
+            ApiError::NoGqlRequest => StatusCode::BAD_REQUEST,
+            ApiError::ParseHttpBody(_) => StatusCode::BAD_REQUEST,
+            ApiError::InvalidWsProtocol => StatusCode::BAD_REQUEST,
+            ApiError::ParseGql(_) => StatusCode::BAD_REQUEST,
+            ApiError::HttpMethodNotAllowed => StatusCode::METHOD_NOT_ALLOWED,
+            ApiError::Auth(_) => StatusCode::UNAUTHORIZED,
+            ApiError::InternalServerError(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        };
+        make_response!(
+            status,
+            json!({ "message": value.to_string(), "success": false })
```

Comment:
I am not sure about the value.to_string() here, i dont think its a good idea to expose the error in the response, since the sql errors might contain sensitive information.

Reply:
What sensitive information could that be? Our db structure is public anyway

Reply:
Oh i see, it seems that all errors other than internal server error are handled

Reply:
When fetching of the db fails, it is a InternalServerError though. Do you think that's a problem?